extension/extensionmiddleware/extensionmiddlewaretest/error_client.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package extensionmiddlewaretest // import "go.opentelemetry.io/collector/extension/extensionmiddleware/extensionmiddlewaretest"
+
+import (
+	"net/http"
+
+	"google.golang.org/grpc"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/extension/extensionmiddleware"
+)
+
+var (
+	_ extension.Extension            = (*errorClient)(nil)
+	_ extensionmiddleware.HTTPClient = (*errorClient)(nil)
+	_ extensionmiddleware.GRPCClient = (*errorClient)(nil)
+)
+
+type errorClient struct {
```

Comment:
I think we use "err" as prefix, see https://github.com/open-telemetry/opentelemetry-collector/blob/main/consumer/consumertest/err.go

Reply:
I copied extensionauthtest verbatim (calls it errorClient) but sure.

Reply:
I see. Thanks. I simplified the code to use the style you showed me.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/pull/12844/commits/5c7fbaaf9021a8b8cca937f510d5348940637f7f